src/main/java/com/redislabs/redisai/Model.java

```diff
@@ -0,0 +1,239 @@
+package com.redislabs.redisai;
+
+import com.redislabs.redisai.exceptions.JRedisAIRunTimeException;
+import java.util.ArrayList;
+import java.util.List;
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.util.SafeEncoder;
+
+/** Direct mapping to RedisAI Model */
+public class Model {
+  private Backend backend;
+  private Device device;
+  private String[] inputs;
+  private String[] outputs;
+  private byte[] blob;
+  private String tag;
+  private long batchSize;
+  private long minBatchSize;
+
+  /**
+   * @param backend - the backend for the model. can be one of TF, TFLITE, TORCH or ONNX
+   * @param device - the device that will execute the model. can be of CPU or GPU
+   * @param inputs - one or more names of the model's input nodes (applicable only for TensorFlow
+   *     models)
+   * @param outputs - one or more names of the model's output nodes (applicable only for TensorFlow
+   *     models)
+   * @param blob - the Protobuf-serialized model
+   */
+  public Model(Backend backend, Device device, String[] inputs, String[] outputs, byte[] blob) {
+    this.backend = backend;
+    this.device = device;
+    this.inputs = inputs;
+    this.outputs = outputs;
+    this.blob = blob;
+    this.tag = null;
+    this.batchSize = 0;
+    this.minBatchSize = 0;
+  }
+
+  /**
+   * @param backend - the backend for the model. can be one of TF, TFLITE, TORCH or ONNX
+   * @param device - the device that will execute the model. can be of CPU or GPU
+   * @param inputs - one or more names of the model's input nodes (applicable only for TensorFlow
+   *     models)
+   * @param outputs - one or more names of the model's output nodes (applicable only for TensorFlow
+   *     models)
+   * @param blob - the Protobuf-serialized model
+   * @param batchSize - when provided with an batchsize that is greater than 0, the engine will
+   *     batch incoming requests from multiple clients that use the model with input tensors of the
+   *     same shape.
+   * @param minBatchSize - when provided with an minbatchsize that is greater than 0, the engine
+   *     will postpone calls to AI.MODELRUN until the batch's size had reached minbatchsize
+   */
+  public Model(
+      Backend backend,
+      Device device,
+      String[] inputs,
+      String[] outputs,
+      byte[] blob,
+      long batchSize,
+      long minBatchSize) {
+    this(backend, device, inputs, outputs, blob);
```

Comment:
this constructor should be called by the other constructor since it has more arguments